src/transformers/tokenization_utils_base.py

```diff
@@ -1809,6 +1817,11 @@ def apply_chat_template(
                 "then to ensure that this model continues working without issues."
             )
 
+        if return_assistant_tokens_mask and not re.search(r"\{\%\s*generation\s*\%\}", chat_template):
+            logger.warning_once(
+                "return_assistant_tokens_mask==True " "but chat template does not contain `{% generation %}` keyword."
+            )
```

Comment:
Good test, but maybe update the regex to also check for `{%-`? You could maybe even just check for the word "generation" at all.